R/external-dependencies.R

```diff
@@ -0,0 +1,111 @@
+external_cli_available <- function(cli) {
+  exit_code <- system(paste("which", cli), ignore.stdout = TRUE, ignore.stderr = TRUE)
+
+  if (exit_code != 0L) {
+    msg <- paste(cli, 'not installed or on $PATH.\n\n')
+    if (cli == "pipx") {
+      msg <- paste0(
+        msg,
+        glue::glue('It can be installed with `install_pipx()`\n\n'),
+        'If already installed, ensure it is on $PATH, e.g. by running',
+        '`pipx ensurepath` or adding `PATH="${PATH}:${HOME}/.local/bin"` to ~/.Renviron'
+      )
+    } else {
+      msg <- paste0(
+        msg,
+        glue::glue('It can be installed with `install_pipx(); install_{cli}()`\n\n'),
+        'If already installed with pipx, ensure it is on $PATH, e.g. by running',
+        '`pipx ensurepath` or adding `PATH="${PATH}:${HOME}/.local/bin"` to ~/.Renviron'
+      )
+    }
+
+    warning(warningCondition(msg, class = "notInstalledWarning"))
+  }
+
+  exit_code == 0L
+}
+
+pipx_available <- function() {
+  external_cli_available(cli = "pipx")
+}
+
+benchconnect_available <- function() {
+  external_cli_available(cli = "benchconnect")
+}
+
+datalogistik_available <- function() {
```

Comment:
I tried messing around with how vscode versus RStudio picks up these paths. Though it is likely something that I've done locally I just wanted to highlight that vscode was only able to pick the paths from the Renviron file. I only raise this because if this got wider usage this issue would likely crop up. Likely fine for now though given limited use of this package for the specific task of arrow R benchmarks.

Reply:
Yeah, to get some of our conversation yesterday where it should be: Mostly this shouldn't be too much of an issue for CI, since everything is run from the command line, but for users running from apps like RStudio, they may need to mess with `PATH` via `.Renviron` or otherwise.

Again, since benchconnect only really needs to be installed where we publish, this probably isn't too much of an issue because that's a lot of machines handled by CI, but for datalogistik, this matters more since benchmark authors will also need it to work locally.